tests/testthat/test-unnecessary_nested_if_linter.R

```diff
@@ -0,0 +1,157 @@
+test_that("unnecessary_nested_if_linter skips allowed usages", {
+  linter <- unnecessary_nested_if_linter()
+
+  expect_lint(
+    trim_some("
+      if (x && y) {
+        1L
+      }"),
+    NULL,
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+      if (x) {
+        1L
+      } else if (y) {
+        2L
+      }"),
+    NULL,
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+      if (x) {
+        1L
+      } else {
+        if (y) {
+          2L
+        }
+      }"),
+    NULL,
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+      if (if (x) TRUE else FALSE) {
+        1L
+      }"),
+    NULL,
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+      if (x) {
+        y <- x + 1L
+        if (y) {
+          1L
+        }
+      }"),
+    NULL,
+    linter
+  )
+
+  expect_lint(
+    trim_some("
+      if ((x && y) || (if (x) TRUE else FALSE)) {
+        1L
+      }"),
+    NULL,
+    linter
+  )
+
+  # if there is any additional code between the inner and outer scopes, no lint
```

Comment:
Stumped by this one; can't seem to find XPath that covers these test cases.

Reply:
Just a hunch, but maybe you need to use `count(*)` somewhere to ensure the first `if ()` only has one child expression?

Reply:
Thanks! That was helpful. Seems to be working now.